.github/workflows/steam.yml

```diff
@@ -0,0 +1,212 @@
+name: Steam Upload
+
+on:
+  release:
+    types:
+    - published
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to fetch and upload (nightly if none)'
+        required: false
+      win_url_override:
+        description: 'Windows build to use (.zip only)'
+        required: false
+      mac_url_override:
+        description: 'Mac build to use (.dmg only)'
+        required: false
+  schedule:
+  - cron: 0 0 * * *
+
+env:
+  WORKFLOW_ID: 583765
+  GIT_NIGHTLY_BRANCH: master
+  STEAM_NIGHTLY_BRANCH: nightly
+  STEAM_STABLE_BRANCH: staging
+  STEAM_BETA_BRANCH: beta_staging
+
+jobs:
+  upload:
+    name: Steam upload
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        path: source
+
+    - name: Setup 7-zip (PPA)
+      # The 7zip version available in the default ubuntu repos (p7zip) is wildly out-of-date and does not properly support DMG files
+      run: |
+        sudo add-apt-repository -y ppa:spvkgn/sevenzip
+        sudo apt update -y
+        sudo apt install -y 7-zip
+
+    - name: Get build information
+      run: |
+        EVENT=${{ github.event_name }}
+        if [[ $EVENT == "release" || ( $EVENT == "workflow_dispatch" && ! -z "${{ github.event.inputs.tag }}") ]]; then
```

Comment:
I always advise to wrap variable names in curly brackets (`${EVENT}`) just as a good practise.

Not quoting the lparam is fine in bash and zsh when using `[[ ]]`. Also if you don't need substitution, I'd use single quotes here.